### DISCUSSION TITLE
Setting OSD (Overlay Text on Screen) via Onvif

### DISCUSSION THREAD

**Discussion by overrideX** [AxisCommunications/discussions#239] [General]
Which cameras support setting/updating OSD via Onvif?
I have tried numerous cameras and none of them seems to have this feature.

**Comment by haydenm315** [comment]
The device requires ONVIF Profile T for OSD.  Out of 13 models of Axis camera in the field of a customer, only the Q6215-LE seems to support Profile T.

**Reply by PhilDay-CT** [reply]
OSD is optional in Profile-S and mandatory in Profile-T.  The API is the same in both cases

**Reply by PhilDay-CT** [reply]
I have found though that you need to be on Axis OS 12.11.77 before it fully works - before that Creating an OSD works but deleting it says its deleted (no longer listed by GetOSDs) but the text stays on the screen, even across a reboot.  The only fix I could find is a factory reset.

A couple of other notes: 
- Only three colours are supported (255,0,0)  (0,0,0) (240,90,82) - The first shows as white although this should be green in YCbRc)
- Onvif OSDs and Axis Overlays are keep separate - you can only manage the Onvif OSDs  via the API
